azure-pipelines.yml

```diff
@@ -73,65 +75,21 @@ strategy:
     #   resolver: "ghc-9.2.2"
     #   stack-yaml: "stack-exact.yaml"
 
-    # just can't get a build-plan at this time. issues with extra,
-    # directory, process & time
-
-    # windows-ghc-master-9.2.2:
-    #   image: "windows-latest"
-    #   mode: "--ghc-flavor ghc-master"
-    #   resolver: "ghc-9.2.2"
-    #   stack-yaml: "stack-exact.yaml"
-
     # +---------+-----------------+------------+
     # | OS      | ghc-lib flavour | GHC        |
     # +=========+=================+============+
-    # | linux   | ghc-master      | ghc-9.0.1  |
-    # | macOS   | ghc-master      | ghc-9.0.1  |
-    # | linux   | ghc-master      | ghc-8.10.7 |
-    # | macOS   | ghc-master      | ghc-8.10.7 |
-    # | windows | ghc-master      | ghc-8.10.7 |
+    # | linux   | ghc-master      | ghc-9.0.2  |
+    # | macOS   | ghc-master      | ghc-9.0.2  |
     # +---------+-----------------+------------+
-    linux-ghc-master-9.0.1:
+    linux-ghc-master-9.0.2:
       image: "ubuntu-latest"
       mode: "--ghc-flavor ghc-master"
-      resolver: "nightly-2021-11-14"
-      stack-yaml: "stack-901.yaml"
-    linux-ghc-master-8.10.7:
-      image: "ubuntu-latest"
-      mode: "--ghc-flavor ghc-master"
-      resolver: "lts-18.16"
+      resolver: "lts-19.8"
       stack-yaml: "stack.yaml"
-    mac-ghc-master-9.0.1:
+    mac-ghc-master-9.0.2:
       image: "macOS-latest"
       mode: "--ghc-flavor ghc-master"
-      resolver: "nightly-2021-11-14"
-      stack-yaml: "stack-901.yaml"
-    mac-ghc-master-8.10.7:
-      image: "macOS-latest"
-      mode: "--ghc-flavor ghc-master"
-      resolver: "lts-18.16"
-      stack-yaml: "stack.yaml"
-    windows-ghc-master-8.10.7:
```

Comment:
macOS and Linux have been cleaned up a bit, but Windows seems to have just disappeared. Was that intentional? If so, would you mind sharing a bit more context on why Windows is no longer desired here?

Reply:
sure. it was intentional. the last remaining windows build was an 8.10.7 and moving to 9.0.2 broke it. for various reasons it's been impossible (for reasons beyond our control) to obtain a windows build since then. i have zero appetite to sink any more hours into it. i am disinclined therefore to continue testing windows builds.

Reply:
note however that ghc-lib builds themselves (as opposed to the production of them) continue to be tested incidentally by ghc-lib-parser-ex and hlint.

Reply:
@garyverhaegen-da

Reply:
It's been a while since I looked at this project; can you double-check if this story I remember is vaguely true?

This project is a Haskell program that packages GHC source code in a way that can be used as a library. The result of running this program is a tarball of Haskell source code that represents a subset of the GHC source tree. Which platform that tarball has been produced on is pretty much irrelevant, since they're just source files.

We keep testing on multiple platforms only to allow for development _on this project_ on multiple platforms, and therefore if there's no one working on this project that uses Windows, running CI on Windows is useless.

The source tarballs can be used on any platform that has a Haskell compiler, so dropping Windows here doesn't prevent e.g. the daml builds from building the daml compiler, using the tarballs produced by running ghc-lib on Linux, from building on Windows.

Is this in the right ballpark?

Reply:
> This project is a Haskell program that packages GHC source code in a way that can be used as a library. The result of running this program is a tarball of Haskell source code that represents a subset of the GHC source tree. Which platform that tarball has been produced on is pretty much irrelevant, since they're just source files.

correct.

> We keep testing on multiple platforms only to allow for development _on this project_ on multiple platforms, and therefore if there's no one working on this project that uses Windows, running CI on Windows is useless.

correct. 

> The source tarballs can be used on any platform that has a Haskell compiler, so dropping Windows here doesn't prevent e.g. the daml builds from building the daml compiler, using the tarballs produced by running ghc-lib on Linux, from building on Windows.

correct.

> 
> Is this in the right ballpark?

yes.